src/components/CommitBox.tsx

```diff
@@ -128,7 +145,10 @@ export class CommitBox extends React.Component<
    * Whether a commit can be performed (files are staged and summary is not empty).
    */
   private _canCommit(): boolean {
-    return !!(this.props.hasFiles && this.state.summary);
```

Comment:
This should not be changed as commit should be available on repository without remote even if the settings `useCommitAndPush` is true => it should fallback on commit only.